packages/@lwc/shared/src/normalize-tab-index.ts

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+
+/**
+ * Global HTML "tabindex" attribute is specially massaged into a stringified number
+ * This follows the historical behavior in api.ts:
+ * https://github.com/salesforce/lwc/blob/f34a347/packages/%40lwc/engine-core/src/framework/api.ts#L193-L211
+ */
+export function normalizeTabIndex(value: any): any {
```

Comment:
```suggestion
export function normalizeTabIndex(value: number): string | number {
```
The original function was more broadly typed than necessary, since this is new we can be more restrictive.

Also, it's weird that this returns either the unmodified number or the string `'0'`. We should either always return a number or always return a string. (It's diverging from the original implementation, but not in a meaningful way.)

Reply:
In the end, the attribute is going to end up a string attribute / numeric prop. This particular function is just an intermediate step in that. This particular function copies the business logic from `engine-core`, but the rest of the implementation (outside the function) is different, so I don't think it makes much of a difference whether the final cast to string/number happens in or out of the function.

Reply:
```suggestion
export function normalizeTabIndex<T>(value: T): T | string {
```
If you want to allow any type.

Reply:
I tried generic but it added issues with the comparison. I wanted to keep it exactly as the original to avoid any discrepancies